include/core/evaporation_model.h

```diff
@@ -0,0 +1,683 @@
+   * of the evaporation recoil pressure with respect to a field
+   * @param field_values Value of the various fields on which the recoil pressure
+   * may depend.
+   * @param id Indicator of the field with respect to which the jacobian
+   * should be calculated
+   * @return value of the partial derivative of the recoil pressure with respect
+   * to the field.
+   */
+  double
+  momentum_flux_jacobian(const std::map<field, double> & /*field_values*/,
+                         const field /*id*/) override
+  {
+    return 0.0;
+  }
+
+  /**
+   * @brief vector_momentum_flux_jacobian Calculate the derivative of the
+   * evaporation recoil pressure with respect to a field
+   * @param field_vectors Vector for the values of the fields on which the recoil
+   * pressure may depend.
+   * @param id Identifier of the field with respect to which a derivative should
+   * be calculated
+   * @param jacobian Vector of the value of the derivative of the recoil pressure
+   * with respect to the field id
+   */
+  void
+  vector_momentum_flux_jacobian(
+    const std::map<field, std::vector<double>> & /*field_vectors*/,
+    const field /*id*/,
+    std::vector<double> &jacobian_vector) override
+  {
+    std::fill(jacobian_vector.begin(), jacobian_vector.end(), 0);
+  }
+
+private:
+  const double n_evaporation;
+  const double latent_heat_evaporation;
+  const double ambient_gas_density_inv;
+  const double liquid_density_inv;
+};
+
+/**
+ * @brief EvaporationModelConstant. Class that allows to calculate the
+ * evaporation heat and momentum fluxes when a constant mass flux is
+ * considered.
+ */
+class EvaporationModelTemperature : public EvaporationModel
+{
+public:
+  EvaporationModelTemperature(const Parameters::Evaporation &p_evaporation)
+    : evaporation_coefficient(p_evaporation.evaporation_coefficient)
+    , recoil_pressure_coefficient(p_evaporation.recoil_pressure_coefficient)
+    , molar_mass(p_evaporation.molar_mass)
+    , boiling_temperature_inv(1.0 / p_evaporation.boiling_temperature)
+    , latent_heat_evaporation(p_evaporation.latent_heat_evaporation)
+    , ambient_pressure(p_evaporation.ambient_pressure)
+    , ambient_gas_density_inv(1.0 / p_evaporation.ambient_gas_density)
+    , liquid_density_inv(1.0 / p_evaporation.liquid_density)
+  {
+    model_depends_on[temperature] = true;
+    this->R_inv                   = 1.0 / 8.3145;
+    this->L_vap_x_M_x_R_inv =
+      latent_heat_evaporation * molar_mass * this->R_inv;
+    this->M_x_2PI_inv_x_R_inv = molar_mass / (2.0 * M_PI) * this->R_inv;
+  }
+
+  /**
+   * @brief saturation_pressure Calculates the value of the staturation pressure.
+   * @param fields_value Value of the various field on which the flux may depend.
+   * @return value of the flux calculated with the fields_value
+   */
+  double
+  saturation_pressure(const std::map<field, double> &fields_value)
+  {
+    const double temperature_inv = 1.0 / fields_value.at(field::temperature);
```

Comment:
I am afraid of division by zero here. Temperature could be zero no.
Maybe add an assertion before?

Reply:
Should I add an assert or add an epsilon to avoid dividing by zero (since exp(-inf)=0):
```suggestion
    const double temperature_inv = 1.0 / (fields_value.at(field::temperature)+1e-12);
```